openlane/scripts/openroad/common/io.tcl

```diff
@@ -316,6 +316,35 @@ proc read_current_odb {args} {
     set_dont_use_cells
 }
 
+proc _populate_cells_by_class {} {
+    if { [info exists ::_cells_by_class(non_core)] } {
+        return
+    }
+
+    set ::_cells_by_class(non_core) [list]
+    set ::_cells_by_class(non_timing) [list]
+    foreach lib $::libs {
+        foreach master [$lib getMasters] {
+            if { "[$master getType]" != "CORE" || "[$master getType]" != "BLOCK" } {
```

Comment:
Can you post the reference behind this classification

Reply:
![metal-gear-rising-metal-gear-rising-revengeance](https://github.com/user-attachments/assets/24378407-bc34-4c52-bd49-836e49223c61)

Reply:
The lefdefref, Macro section

CORE just means a macro used in the core area (i.e. a standard cell.)

Thing is, it has a lot of subclasses for the non-logical cells:

`FEEDTHRU`,`SPACER`,`ANTENNACELL`,`WELLTAP`.

(Also, `TIEHIGH` and `TIELOW` but they appear to be unused in the PDKs I checked.)

I naively assume anything that is CORE and not further subclassed is a logical cell. This is not comprehensive, but removes most of the fills and such which do not contribute to the logical functionality of the circuit.

I do make one exception for `ANTENNACELL`s. These are not counted as logical cells but they are not exempt from the so-called SDF-friendly netlist as they do affect timing ever so slightly.

`BLOCK` is for macros. We need to keep those for simulations.